MassMessage/index.ts

```diff
@@ -16,47 +17,55 @@ const timerTrigger: AzureFunction = async function (context: Context, myTimer: a
 
     await MongoConnect()
 
-    const date = new Date()
+    const curDate = new Date()
     //avoiding the use of times
-    date.setHours(0)
-    date.setMinutes(0)
-    date.setSeconds(0)
-    date.setMilliseconds(0)
+    curDate.setHours(0)
+    curDate.setMinutes(0)
+    curDate.setSeconds(0)
+    curDate.setMilliseconds(0)
+    // passing this date around so we don't need to repeatedly set times to 0
 
-    //set to two months ago
-    date.setMonth(date.getMonth() - 2)
-
-    const inactive = await messageInactiveUsers(date)
+    const inactive = await messageInactiveUsers(curDate)
     context.log(inactive)
-    const catchup = await messageCompletedUsers(date, context)
+    const catchup = await messageCompletedUsers(curDate, context)
     context.log(catchup)
 }
 
-//gets users where last activity was exactly 2 months ago
+//gets users where last activity was exactly 2 weeks ago
 const messageInactiveUsers = async function (date: Date) {
+    //set to two weeks ago
+    const twoWeeks = new Date(date);
+    twoWeeks.setDate(twoWeeks.getDate() - 14);
+    
     const allUsers = await UserState.find({
-        lastActivity: date,
+        lastActivity: twoWeeks,
     })
     allUsers.forEach((user) => {
-        sendInactiveMessage(user.userId, date.toDateString())
+        sendInactiveMessage(user.userId, twoWeeks.toDateString(), user.lowData)
+        //hardcoded to welcome message
+        user.currMessage = Mongoose.Types.ObjectId("6022178429efc055c8e74e50");
     })
 }
 
 //gets users where they completed a module 2 months ago
 const messageCompletedUsers = async function (date: Date, context: Context) {
-    const prevDay = new Date(date.toDateString())
+    //set to two months ago
+    const twoMonths = new Date(date);
+    twoMonths.setMonth(twoMonths.getMonth() - 2);
+
+    const prevDay = new Date(twoMonths)
     prevDay.setDate(prevDay.getDate() - 1)
 
     const allUsers = await UserState.find({
         moduleCompletionTime: {
-            $gte: prevDay,
-            $lt: date,
+            $gt: prevDay,
+            $lt: twoMonths,
```

Comment:
Very likely the reason why the query doesn't work is because we're doing operations on an array, not an individual value. As such, we need to query for if there is a match within this range for any value in this array, rather than if the array is greater than or less than a date, which doesn't make much sense. The elemmatch pipe will likely be helpful for this.
https://docs.mongodb.com/manual/tutorial/query-arrays/